src/mlpack/methods/cf/cf_main.cpp

```diff
@@ -142,17 +161,139 @@ void ComputeRecommendations(CFModel* cf,
 
     Log::Info << "Generating recommendations for " << users.n_elem << " users."
         << endl;
-    cf->GetRecommendations(numRecs, recommendations, users.row(0).t());
+
+    //  Making All the possible paths
+    if (ns_algo == "Cosine")
```

Comment:
Please use camel casing for all names; for more information take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions.